order/orderRepo.py

```diff
@@ -0,0 +1,205 @@
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+from .orderCRUD import OrderCreate, OrderItemCreate, OrderItemResponse, OrderItemUpdate, OrderResponse, OrderUpdate
+from .orderModels import Order, Order_Item, StatusEnum
+
+
+
+
+# ***********      Order     ************
+
+class OrderRepository:
+    def __init__(self, session: Session):
+        self.session = session
+        
+    #   CREATE
+    def create_new_order(self, order: OrderCreate):
+        new_order = Order(
+            
+            user_id = order.user_id,
+            shipping_address = order.shipping_address, 
+            billing_address = order.billing_address,
+            status = order.status
+        )
+        
+        
+        # Session aktualisieren
+        self.session.add(new_order)
+        self.session.commit()
+        self.session.refresh(new_order)
+        
+        
+        order_response = self.create_order_response(new_order)
+
+        return order_response
+    
+    
+    
+    def create_order_response(self, order: Order) -> OrderResponse:
+
+        items = [
+            OrderItemResponse(
+                unique_order_item_id=item.unique_order_item_id,
+                product_id=item.product_id,
+                quantity=item.quantity,
+                price=item.price,
+            )
+            for item in order.items
+        ]
+        
+        total_price = sum(item.price * item.quantity for item in order.items)
+
+
+        response = OrderResponse(
+            unique_order_id=order.unique_order_id,
+            user_id=order.user_id,
+            created_at=order.created_at,
+            shipping_address=order.shipping_address,
+            billing_address=order.billing_address,
+            total_price=total_price,  
+            status=order.status,
+            items=items,
+        )
+        
+        return response
+
+    
+    
+    # READ
+    def get_by_order_id(self, unique_order_id: int):
+        stmt = select(Order).where(Order.unique_order_id == unique_order_id)
+        result = self.session.execute(stmt).scalars().first()
+        
```

Comment:
In das repo kommen nur Datenbankinteraktionen rein.
Dir fehlt der Service, z.B. zeile 73-77 müsste in den service.